staging/src/k8s.io/apiserver/pkg/server/filters/timeout_test.go

```diff
@@ -217,11 +223,11 @@ func captureStdErr() (func() string, func(), error) {
 	}
 	klog.LogToStderr(true)
 	cleanUp := func() {
-		os.Stderr = stderr
+		safeSetStderr(stderr)
 		klog.LogToStderr(false)
```

Comment:
The log in the linked issue showed that the race was between this test setting os.Stderr and klog calling to os.Stderr.Write.

So if we run `klog.LogToStderr(false)` first, before running `os.Stderr = stderr`, does that solve the data race? That's a cleaner fix.

Reply:
Thank you, seems feasible.